### DISCUSSION TITLE
[PSD13] Slide 2 - Esercizio 1

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/psd2223#187] [Esercizi (A-L)]
1) Trasformare la macchina a stati di Moore del semaforo da Moore a Mealy.

Per "semaforo" l'esercizio a cui si riferisce è molto probabilmente l'**Esercizio 3.24** (#184 )

**Comment by Elia-Belli** [comment]
La macchina del semaforo a cui ci riferiamo è la versione **originale** dell' **Esercizio 3.24** (https://github.com/sapienzastudentsnetwork/psd2223/discussions/184#discussioncomment-9052109)

![SD](https://user-images.githubusercontent.com/107944476/215059392-0b21a52d-8046-4685-a51f-824f111f12aa.jpg)

**State Encodings** : $S0=0;\ S1=1;$
**Output Encodings**: Red: $00$; Green: $01$; Yellow: $10$;

| $S$ | $T_A$ | $T_B$ | $S'$ | $LA_1$ | $LA_0$ | $LB_1$ | $LB_0$ | 
|---|---|---|---|---|---|---|---|
|0|0|X|1|1|0|0|0|
|0|1|X|0|0|1|0|0|
|1|X|0|0|0|0|1|0|
|1|X|1|1|0|0|0|1|

$S' = \bar S \bar T_A + S T_B$
$LA_1 = \bar S \bar T_A$
$LA_0 = \bar S T_A$
$LB_1 = S \bar T_B$
$LB_0 = S T_B$

![SC](https://user-images.githubusercontent.com/107944476/215059418-d3591de7-d4a9-4d3b-806f-1335849b5c6b.jpg)

**Comment by FeddyLix17** [comment]
tutto corretto

**Comment by Amirhossein-2020** [comment]
Grazie, ma ho un problema con questo diagramma, il che è nella domanda è detto che la luce diventa gialla per 5 secondi prima che l'altra diventi verde, ma con questo diagramma possa accadere che la seconda luce diventi verde subito saltando lo stato di mezzo.

**Comment by FeddyLix17** [comment]
@Amirhossein-2020 lo stato di **mezzo** di cui parli si verifica rispettivamente quando T<sub>A</sub> = 0 e T<sub>B</sub> = 0, in cui si ritorniamo allo stato corrispondente a seconda del caso ma con un output differente (giallo = 10), usando mealy non c'è stato bisogno di creare 2 stati appositi per rappresentare quelle transizioni (sempre considerando che  in moore dopo quei stati indipendentemente dagli input saremmo ritornati in S<sub>0</sub> o S<sub>1</sub>)
